tuta-sdk/rust/sdk/src/json_serializer.rs

```diff
@@ -117,11 +134,14 @@ impl JsonSerializer {
                     JsonElement::String(id),
                 ) => {
                     // FIXME it's not always generated id but it's fine probably
-                    mapped.insert(association_name, ElementValue::IdGeneratedId(GeneratedId(id)));
+                    mapped.insert(
+                        association_name,
+                        ElementValue::IdGeneratedId(GeneratedId(id)),
+                    );
                 }
                 (
                     AssociationType::ListElementAssociation,
-                    Cardinality::One,
+                    Cardinality::One | Cardinality::ZeroOrOne,
```

Comment:
For `ListElementAssociation`, `ZeroOrOne` is not `Option`?

Reply:
what do you mean? We already handle all `ZeroOrOne` + `Null` cases together?

Reply:
Ah ok I got confused. There are too many conditions.